key-rotator/storage/key_aws.go

```diff
@@ -0,0 +1,103 @@
+package storage
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/aws/request"
+	"github.com/aws/aws-sdk-go/service/secretsmanager"
+	"github.com/rs/zerolog/log"
+
+	"github.com/abetterinternet/prio-server/key-rotator/key"
+)
+
+// NewAWSKey returns a Key implementation using the AWS secret manager for
+// backing storage. This key store writes keys in a way that is suitable for
+// backup; keys written by this store cannot be read by other components of the
+// Prio system (e.g. the facilitator).
+func NewAWSKey(sm *secretsmanager.SecretsManager, prioEnv string) Key { return awsKey{sm, prioEnv} }
+
+type awsKey struct {
```

Comment:
Should there be a `var _ awsKey = Key` to verify that this satisfies the `Key` interface?